### PR TITLE
Add instructions for making block attribute required with custom validator

### PR DESCRIPTION
**Pull request by ghost**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | 
| Versions      | master, 4.6, 4.5
| Edition       | 

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

**Review by Steveb-p — APPROVED** [review]
Using form extension to work with custom options is the way to go :+1:

Note that this applies to Attribute Definitions, which as I understand it is slightly different from Attributes. Attribute Definition is a whole group which represents an Attribute (`value`), including it's `name` and `id`.

See `Ibexa\PageBuilder\Form\Type\Attribute\AttributeType::buildForm()`, which in turn creates specific Attribute value types (using a mapper).

To make this easier on developers (we should gauge the need for it), we could expose a configuration option to add more constraints that should be treated as marking the form field as required :thinking:

**Comment by ms2ad** [comment]
@Steveb-p I see what you mean that whether using the whole `$builder` object might be the right case because all fields are children of it, including the name and id field, not just the value.

> To make this easier on developers (we should gauge the need for it), we could expose a configuration option to add more constraints that should be treated as marking the form field as required 🤔

Your current implementation might also benefit from it because while having two specific validators hard-coded in `AttributeType::buildForm()` works, but is not the most clean code imaginable :)

**Review by mnocon — APPROVED** [review]
Thank you for the contribution! 🙇

**Review by dabrt — APPROVED** [review]

File: docs/content_management/pages/page_block_validators.md [hunk 1]
```diff
@@ -66,3 +66,12 @@ Finally, add the validator to one of your block attributes, for example:
 ``` yaml hl_lines="16-18"
 [[= include_file('code_samples/page/custom_block_validator/config/packages/page_blocks.yaml', 0, 1) =]][[= include_file('code_samples/page/custom_block_validator/config/packages/page_blocks.yaml', 3, 20) =]]
 ```
+
+### Custom required validator
+
+By default, only `not_blank` and `not_blank_richtext` cause a block attribute to be marked as required.
+If you create a custom validator `custom_not_blank` with attribute-specific logic, you can add extend the `AttributeType` class with a Symfony form type extension to ensure that the attribute is also considered required:
+
+``` php
```
Comment (adriendupuis) [inline docs/content_management/pages/page_block_validators.md:75]: I would highlight `$builder->setRequired(true);`:
```suggestion
``` php hl_lines="16"
```

File: docs/content_management/pages/page_block_validators.md [hunk 2]
```diff
@@ -66,3 +66,12 @@ Finally, add the validator to one of your block attributes, for example:
 ``` yaml hl_lines="16-18"
 [[= include_file('code_samples/page/custom_block_validator/config/packages/page_blocks.yaml', 0, 1) =]][[= include_file('code_samples/page/custom_block_validator/config/packages/page_blocks.yaml', 3, 20) =]]
 ```
+
+### Custom required validator
+
+By default, only `not_blank` and `not_blank_richtext` cause a block attribute to be marked as required.
```
[the remark this answers is not in the document]
Comment (adriendupuis) [reply docs/content_management/pages/page_block_validators.md:72]: Nice to have a list of validators automatically triggering the "required" mark. It justifies a new section by itself. I would also add a new paragraph separator:

```suggestion
By default, only `not_blank` and `not_blank_richtext` validators mark a block attribute as required.

```

**Review by adriendupuis — CHANGES_REQUESTED** [review]
~~See review comments~~ ✅ applied

**Review by adriendupuis — APPROVED** [review]

File: docs/content_management/pages/page_block_validators.md [hunk 3]
```diff
@@ -66,3 +66,13 @@ Finally, add the validator to one of your block attributes, for example:
 ``` yaml hl_lines="16-18"
 [[= include_file('code_samples/page/custom_block_validator/config/packages/page_blocks.yaml', 0, 1) =]][[= include_file('code_samples/page/custom_block_validator/config/packages/page_blocks.yaml', 3, 20) =]]
 ```
+
+### Custom required validator
+
+By default, only `not_blank` and `not_blank_richtext` validators mark a block attribute as required.
+
+If you create a custom validator `custom_not_blank` with attribute-specific logic, you can extend the `AttributeType` class with a Symfony form type extension to ensure that the attribute is also considered required:
```
Comment (adriendupuis) [inline docs/content_management/pages/page_block_validators.md:74]: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/ensure-insure-assure
```suggestion
If you create a custom validator `custom_not_blank` with attribute-specific logic, you can extend the `AttributeType` class with a Symfony form type extension to make sure that the attribute is also considered required:
```

**Comment by adriendupuis** [comment]
@ms2ad Thank you!

**Pull request merged**